pyproject.toml

```diff
@@ -1,16 +1,65 @@
 [build-system]
-# Minimum requirements for the build system to execute.
 requires = [
-        "setuptools>=42.0.0",
-        "wheel>=0.34.2",
-        "numpy;python_version >= '3.9' and platform_python_implementation == 'PyPy'",
-        "oldest-supported-numpy;python_version < '3.9' or platform_python_implementation != 'PyPy'",
-        "scikit-build",
-        "cmake>=3.17",
-        "nanobind>=1.9.2",
-        "ninja; platform_system!='Windows'",
-        ]
-build-backend = "setuptools.build_meta"
+    "scikit-build-core >=0.9.3",
+    "nanobind >=1.9.2",
+    "numpy;python_version >= '3.9' and platform_python_implementation == 'PyPy'",
+    "oldest-supported-numpy;python_version < '3.9' or platform_python_implementation != 'PyPy'",
+]
+build-backend = "scikit_build_core.build"
+
+[tool.scikit-build.cmake.define]
+PYOPENCL_TRACE = {env="CL_TRACE", default="OFF"}
+PYOPENCL_ENABLE_GL = {env="CL_ENABLE_GL", default="OFF"}
+PYOPENCL_USE_SHIPPED_EXT = {env="CL_USE_SHIPPED_EXT", default="OFF"}
+OpenCL_INCLUDE_DIR = {env="CL_INC_DIR"}
+PYOPENCL_CL_LIB_DIRS = {env="CL_LIB_DIR"}
+OpenCL_LIBRARY = {env="CL_LIBNAME"}
```

Comment:
These should also be defined in `CMakeLists.txt` as `option(PYOPENCL_TRACE "Enable tracing functionality" OFF)`? At least that's idiomatic from what I recall..

Reply:
Done in [34a47d9](https://github.com/inducer/pyopencl/pull/755/commits/34a47d97e355791c8dfbf99174a0da168124482a)